src/reducers/index.ts

```diff
@@ -0,0 +1,38 @@
+import ICounter from '../types';
+
+const initialState = [
+  { count: 0 },
+  { count: 5 }
+];
+
+interface ICounterReducerAction {
```

Comment:
I believe this interface should exist in an action file. The other benefit of having an action file is that you can define your actions as functions that return the appropriate action type. This removes the need to always specify the action type and makes your code less "stringly typed". For example here are a few places that having an action creator function would remove "stringly typed" code:

https://github.com/maclover7/learning-project/pull/2/files#diff-376eface0cf686f74f59162a30da9626R5
https://github.com/maclover7/learning-project/pull/2/files#diff-376eface0cf686f74f59162a30da9626R11
https://github.com/maclover7/learning-project/pull/2/files#diff-46ef2374da6211486f3a408db68b2bcbR44
https://github.com/maclover7/learning-project/pull/2/files#diff-46ef2374da6211486f3a408db68b2bcbR45
https://github.com/maclover7/learning-project/pull/2/files#diff-0c230c2c950beb1f5987a2aed0eeeab3R15
https://github.com/maclover7/learning-project/pull/2/files#diff-0c230c2c950beb1f5987a2aed0eeeab3R25

Here is how I solve this issue in Live:
Defining the Action types: https://github.com/showclix/Live-React-Native/blob/dev/src/actions/device.ts#L4
Creating the Action creators: https://github.com/showclix/Live-React-Native/blob/dev/src/actions/device.ts#L23
(Ignore the wrapping of it in createActions, that is only for the strange type stuff I'm doing. It can just return an object.)
Checking for Action Type in a reducer: https://github.com/showclix/Live-React-Native/blob/dev/src/reducers/device.ts#L19
Binding the Action Creators in a component: https://github.com/showclix/Live-React-Native/blob/dev/src/containers/eventDetailScreen.tsx#L337
(I bind all of Action Creators to an actions object so to call them you have to use `props.actions.<whatever>`, this is just a personal choice of mine. I don't expect you to do it in the learning project.)